src/shaderlib/project/project.glsl.js

```diff
@@ -73,45 +83,33 @@ vec4 project_scale(vec4 meters) {
 // Projecting positions
 //
 
-// non-linear projection: lnglats => unit tile [0-1, 0-1]
-vec2 project_mercator_(vec2 lnglat) {
-  return vec2(
-    radians(lnglat.x) + PI,
-    PI - log(tan_fp32(PI * 0.25 + radians(lnglat.y) * 0.5))
-  );
-}
-
-vec2 project_position(vec2 position) {
-  if (projectionMode == PROJECT_IDENTITY) {
-    return position;
-  }
-  if (projectionMode == PROJECT_MERCATOR_OFFSETS) {
-    return project_scale(position);
-  }
-  // Covers projectionMode == PROJECT_MERCATOR
-  return project_mercator_(position) * WORLD_SCALE * projectionScale;
+vec4 project_position(vec4 position) {
+  // Apply model matrix
+  vec4 position_modelspace = modelMatrix * position;
+  return project_scale(position_modelspace);
 }
 
 vec3 project_position(vec3 position) {
-  return vec3(project_position(position.xy), project_scale(position.z));
+  vec4 projected_position = project_position(vec4(position, 1.0));
+  return projected_position.xyz;
 }
 
-vec4 project_position(vec4 position) {
-  return vec4(project_position(position.xyz), position.w);
+vec2 project_position(vec2 position) {
+  vec4 projected_position = project_position(vec4(position, 0.0, 1.0));
+  return projected_position.xy;
 }
 
-//
-
+// TODO/xiaoji: this function no longer produces correct result, remove?
```

Comment:
With the changes in this PR this function no longer accepts raw positions, which defeats the original purpose of adding it. It also gives incorrect result in METER_OFFSETS mode. 

we don't use this function anywhere except in the plot layer example. As shown above, I am able to make the plot layer work without it. At this point it's too much trouble fixing it and it doesn't provide a lot of value.